src/App.tsx

```diff
@@ -1,223 +1,256 @@
+import {
+  createTodo, getTodos, patchTodo, removeTodo,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { TodosSelections } from './types/TodosSelections';
+import { ErrorHandler } from './types/ErrorHandler';
+import { ErrorNotification } from './components/ErrorNotification';
+import { TodoHeader } from './components/TodoHeader';
 
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [hasError, setHasError] = useState<ErrorHandler>(ErrorHandler.None);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
+  const [filterSelector, setFilterSelector]
+    = useState<TodosSelections>(TodosSelections.All);
+  const [isAdding, setIsAdding] = useState(false);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [todosChangedIds, setTodosChangedIds]
+    = useState<number[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo>({
+    id: 0,
+    userId: 0,
+    title: '',
+    completed: false,
+  });
+  const isTodosAvailable = todos.length > 0;
+
+  const closeErrorField = () => (
+    setHasError(ErrorHandler.None)
+  );
+
+  const remainCompletedTodos = useMemo(() => (
+    todos.filter(todo => todo.completed).length
+  ), [todos]);
+
+  const getTodosFromServer = useCallback(
+    async () => {
+      try {
+        if (user) {
+          const todosFromServer = await getTodos(user.id);
+
+          setTodos(todosFromServer);
+          setFilteredTodos(todosFromServer);
+        }
+      } catch (error) {
+        setHasError(ErrorHandler.LoadError);
+      }
+    }, [],
+  );
+
+  const addTodoToServer = useCallback(
+    async () => {
+      try {
+        if (!todoTitle.trim()) {
+          setHasError(ErrorHandler.EmptyTitle);
+
+          return;
+        }
+
+        setIsAdding(true);
+
+        const addingData = {
+          title: todoTitle,
+          userId: user?.id || 0,
+          completed: false,
+        };
+
+        setTempTodo((currentTodo) => ({
+          ...currentTodo,
+          ...addingData,
+        }));
+
+        await createTodo(addingData);
+        await getTodosFromServer();
+
+        setIsAdding(false);
+        setTodoTitle('');
+      } catch (error) {
+        setIsAdding(false);
+        if (error === ErrorHandler.EmptyTitle) {
+          setHasError(ErrorHandler.EmptyTitle);
+        } else {
+          setHasError(ErrorHandler.AddError);
+        }
+      }
+    }, [todoTitle],
+  );
+
+  const removeTodoFromServer = useCallback(
+    async (todoId: number) => {
+      try {
+        setTodosChangedIds(prevIds => [...prevIds, todoId]);
+        await removeTodo(todoId);
+        await getTodosFromServer();
+      } catch (error) {
+        setHasError(ErrorHandler.DeleteError);
+      }
+    }, [todos],
+  );
```

Comment:
i am not sure, but maybe you need to make functions like this not in the App component